docs/src/main/asciidoc/security-oidc-code-flow-authentication-concept.adoc

```diff
@@ -267,285 +660,167 @@ quarkus.oidc.logout.post-logout-uri-param=returnTo
+[[session-management]]
 
-If you prefer to use SPA and JavaScript API such as `Fetch` or `XMLHttpRequest`(XHR) with Quarkus web applications, be aware that OpenID Connect Providers may not support CORS for Authorization endpoints where the users are authenticated after a redirect from Quarkus. This will lead to authentication failures if the Quarkus application and the OpenID Connect Provider are hosted on the different HTTP domains/ports.
+==== Expiration via Session management
 
-In such cases, set the `quarkus.oidc.authentication.java-script-auto-redirect` property to `false` which will instruct Quarkus to return a `499` status code and `WWW-Authenticate` header with the `OIDC` value. The browser script also needs to be updated to set `X-Requested-With` header with the `JavaScript` value and reload the last requested page in case of `499`, for example:
+By default, the logout is based on the expiration time of the ID Token issued by the OpenID Connect Provider. When the ID Token expires, the current user session at the Quarkus endpoint is invalidated and the user is redirected to the OpenID Connect Provider again to authenticate. If the session at the OpenID Connect Provider is still active, users are automatically re-authenticated without having to provide their credentials again.
 
-[source,javascript]
-----
-Future<void> callQuarkusService() async {
-    Map<String, String> headers = Map.fromEntries([MapEntry("X-Requested-With", "JavaScript")]);
+The current user session may be automatically extended by enabling a `quarkus.oidc.token.refresh-expired` property. If it is set to `true` then when the current ID Token expires a Refresh Token Grant will be used to refresh ID Token as well as Access and Refresh Tokens.
 
-    await http
-        .get("https://localhost:443/serviceCall")
-        .then((response) {
-            if (response.statusCode == 499) {
-                window.location.assign("https://localhost.com:443/serviceCall");
-            }
-         });
-  }
-----
+[TIP]
+--
+If you have a xref:security-oidc-bearer-token-authentication-concept.adoc#single-page-applications[Single Page Application for Service Applications] where your OpenID Connect Provider script such as `keycloak.js` is managing an authorization code flow then that script will also control the SPA authentication session lifespan.
+--
 
-=== Cross-origin resource sharing
+If you work with a Quarkus OIDC `web-app` application then it is Quarkus OIDC Code Authentication mechanism which is managing the user session lifespan.
 
-If you plan to consume this application from a Single-page application running on a different domain, you will need to configure CORS (Cross-Origin Resource Sharing). For more information, see the xref:http-reference.adoc#cors-filter[HTTP CORS documentation].
+To use the refresh token, you should carefully configure the various lifespan of the ID token, the refresh token, the cookie, etc so they work in tandem.
```

Comment:
```suggestion
To use the refresh token, you should carefully configure the various lifespan of the ID token, the refresh token, the cookie, and any other elements needed, so that they work in tandem.
```

Reply:
I'd like to review this sentence, on its own it does not make a lot of sense (meaning I likely created it in the first place :-) )

Reply:
@sberyozkin - No problem.  Let me know if you need my input on the rewrite. Thanks.

Reply:
@michelle-purcell Thanks, for now, I simplified it to `To use the refresh token, you should carefully configure session cookie age.` with the instructions following